cmd/crates/soroban-spec-typescript/src/project_template/src/assembled-tx.ts

```diff
@@ -0,0 +1,706 @@
+      throw e
+    }
+  }
+
+  parseError(errorMessage: string): Err | undefined {
+    if (!this.options.errorTypes) return
+    const match = errorMessage.match(contractErrorPattern)
+    if (!match) return
+    let i = parseInt(match[1], 10)
+    let err = this.options.errorTypes[i]
+    if (err) return new Err(err)
+  }
+
+  /**
+   * Get account details from the Soroban network for the publicKey currently
+   * selected in user's wallet. If not connected to Freighter, use placeholder
+   * null account.
+   */
+  getAccount = async (): Promise<Account> => {
+    const wallet = await getWallet(this.options.wallet)
+
+    let publicKey: string | undefined
+    if (await wallet.isConnected() && await wallet.isAllowed()) {
+      publicKey = (await wallet.getUserInfo()).publicKey
+    }
+
+    return publicKey
+      ? await this.server.getAccount(publicKey)
+      : new Account(NULL_ACCOUNT, "0")
+  }
+
+  /**
+   * Sign the transaction with the `wallet` (default Freighter), then send to
+   * the network and return a `SentTransaction` that keeps track of all the
+   * attempts to send and fetch the transaction from the network.
+   */
+  signAndSend = async ({ secondsToWait = 10, force = false }: {
+    /**
+     * Wait `secondsToWait` seconds (default: 10) for both the transaction to SEND successfully (will keep trying if the server returns `TRY_AGAIN_LATER`), as well as for the transaction to COMPLETE (will keep checking if the server returns `PENDING`).
+     */
+    secondsToWait?: number
+    /**
+     * If `true`, sign and send the transaction even if it is a read call.
+     */
+    force?: boolean
+  } = {}): Promise<SentTransaction<T>> => {
+    if (!this.raw) {
+      throw new Error('Transaction has not yet been simulated')
+    }
+
+    if (!force && this.isReadCall) {
+      throw new Error('This is a read call. It requires no signature or sending. Use `force: true` to sign and send anyway.')
+    }
+
+    if (!await this.hasRealInvoker()) {
+      throw new WalletDisconnectedError('Wallet is not connected')
+    }
+
+    if (this.raw.source !== (await this.getAccount()).accountId()) {
+      throw new Error(`You must submit the transaction with the account that originally created it. Please switch to the wallet with "${this.raw.source}" as its public key.`)
+    }
+
+    if ((await this.needsNonInvokerSigningBy()).length) {
+      throw new NeedsMoreSignaturesError(
+        'Transaction requires more signatures. See `needsNonInvokerSigningBy` for details.'
+      )
+    }
+
+    return await SentTransaction.init(this.options, this, secondsToWait);
+  }
+
+  getStorageExpiration = async (storageType: 'temporary' | 'persistent') => {
+    const key = xdr.LedgerKey.contractData(
+      new xdr.LedgerKeyContractData({
+        contract: new Address(this.options.contractId).toScAddress(),
+        key: xdr.ScVal.scvLedgerKeyContractInstance(),
+        durability: xdr.ContractDataDurability[storageType](),
+      }),
+    )
```

Comment:
you can use [`getFootprint`](https://stellar.github.io/js-soroban-client/Contract.html#getFootprint) here (see the [code](https://stellar.github.io/js-soroban-client/js-stellar-base_src_contract.js.html#line78)):

```suggestion
    const key = new Contract(this.options.contractId).getFootprint()[1];
```

The instance is always persistent, afaik.

(note this will need a [small amendment](https://github.com/stellar/js-stellar-base/pull/709) after the stable release)